libimage/manifest_list.go

```diff
@@ -221,17 +222,67 @@ func (i *Image) IsManifestList(ctx context.Context) (bool, error) {
 // Inspect returns a dockerized version of the manifest list.
 func (m *ManifestList) Inspect() (*define.ManifestListData, error) {
 	inspectList := define.ManifestListData{}
+	// Copy the fields from the Docker-format version of the list.
 	dockerFormat := m.list.Docker()
 	err := structcopier.Copy(&inspectList, &dockerFormat)
 	if err != nil {
 		return &inspectList, err
 	}
-	// Get missing annotation field from OCIv1 Spec
-	// and populate inspect data.
+	// Get OCI-specific fields from the OCIv1-format version of the list
+	// and copy them to the inspect data.
 	ociFormat := m.list.OCIv1()
+	inspectList.ArtifactType = ociFormat.ArtifactType
 	inspectList.Annotations = ociFormat.Annotations
 	for i, manifest := range ociFormat.Manifests {
 		inspectList.Manifests[i].Annotations = manifest.Annotations
+		inspectList.Manifests[i].ArtifactType = manifest.ArtifactType
+		inspectList.Manifests[i].URLs = append([]string{}, manifest.URLs...)
```

Comment:
(`slices.Copy` can be used for copies like this (throughout).)

Reply:
Ooh, only learned about this package from a linter in the last couple of days.  Will fix.